x-pack/plugins/apm/server/lib/helpers/setup_request.ts

```diff
@@ -104,3 +107,25 @@ export async function setupRequest<TParams extends SetupRequestParams>(
     ...coreSetupRequest,
   } as InferSetup<TParams>;
 }
+
+function getMlSetup(context: APMRequestHandlerContext, request: KibanaRequest) {
+  if (!context.plugins.ml) {
+    return;
+  }
+  const ml = context.plugins.ml;
+  const mlClient = ml.mlClient.asScoped(request).callAsCurrentUser;
+  return {
+    ...ml.mlSystemProvider(mlClient, request),
+    mlClient,
+    /**
+     * https://www.elastic.co/guide/en/elasticsearch/reference/7.x/ml-get-job.html#ml-get-job-desc
+     * @param {string | string[]} [jobId] - job id, group name, or a wildcard, returns all jobs if nothing passed in
+     */
+    mlJobs: async (
+      jobId?: string | string[]
+    ): Promise<AnomalyDetectionJob[]> => {
+      const mlJobsResponse = await mlClient('ml.jobs', { jobId });
```

Comment:
To comply with our current spaces work, external plugins should no long be calling the ML elasticsearch endpoints directly. 
They need to instead be calling the functions shared in out setup contract.
This call is covered by `jobs` inside `anomalyDetectorsProvider`.

Note, I just noticed that the return type for `jobs` is using an `any` rather than the correct return return type of  `{ count: number; jobs: Job[] }`
This will be corrected very soon as part of our on going spaces work.

Reply:
in the latest commit i updated the return type of that API within ML